metaflow/plugins/kubernetes/kubernetes_decorator.py

```diff
@@ -98,9 +105,17 @@ def __init__(self, attributes=None, statically_defined=False):
             self.attributes["node_selector"] = KUBERNETES_NODE_SELECTOR
         if not self.attributes["tolerations"] and KUBERNETES_TOLERATIONS:
             self.attributes["tolerations"] = json.loads(KUBERNETES_TOLERATIONS)
+        if not self.attributes["labels"] and KUBERNETES_LABELS:
+            self.attributes["labels"] = KUBERNETES_LABELS
+
+        if isinstance(self.attributes["labels"], str):
+            self.attributes["labels"] = self.parse_kube_keyvalue_list(
+                self.attributes["labels"].split(","), False
+            )
+        self.attributes["labels"] = self.validate_kube_labels(self.attributes["labels"])
```

Comment:
Nit: relies on `validate_kube_labels` to return something, which given its name is unexpected. As the validator is more concerned about raising exceptions, it doesn't necessarily need to return anything. Could we instead simply call it to validate the labels in the attributes?

Based on the [lifecycle graph](https://github.com/Netflix/metaflow/blob/master/docs/lifecycle.png) , one place to call this could be in the `step_init()` phase

not a dealbreaker though so feel free to ignore this, as the current implementation also works :)

Reply:
So I changed the function to return a bool or throw an exception.